File: links.pl
#!/usr/bin/perl
use HTTP::Tiny;
use Data::Dumper::Perltidy;

$message="Hello, Welcome to LinuxHint";
print("$message \n");

#use warnings;
#use HTTP::Request; use HTTP::Headers; use LWP::UserAgent; use JSON::PP;

$apiKEY="?apikey=3e00873e2f8047a4a541d8d11b439330";
$host="https://192.168.1.100:8989/api/v3";
$targetAPI="history";

$SERVER_ADDR="http://192.168.1.100:8989/api/v3/history?apikey=3e00873e2f8047a4a541d8d11b439330";




#use LWP::Simple;
#$content = get($url);
#die "Can't GET $url" if (! defined $content);

my $response = HTTP::Tiny->new->get($SERVER_ADDR);
if ($response->{success})
{
    my $html = $response->{content};
    @LINES = split /\n/, $html;
    chomp(@LINES);
    #print("Lines: '@LINES'\n"); # ZZZ
    my $decoded_json = decode_json( $html );
    print Dumper $decoded_json;
}
else
{
    print "Failed: $response->{status} $response->{reasons}";
}

File: links.py
#!/usr/bin/python3
print('hello world')

import requests
import json
import pprint


response = requests.get('http://192.168.1.100:8989/api/v3/history?apikey=3e00873e2f8047a4a541d8d11b439330') 
#json_data = json.loads(response.text)
jsonResponse = response.json()

#pprint.pprint(json_data)

# change the JSON string into a JSON object
#jsonObject = json.loads(jsonResponse)

# print the keys and values
#for key in jsonResponse:
#  value = jsonResponse[key]
  
pprint.pprint(jsonResponse.records.data[0])
